internal/receiver/smartagentreceiver/log_test.go

```diff
@@ -187,6 +202,144 @@ func TestRedirectShouldUniquelyAddHooksToLogrusKeyLogger(t *testing.T) {
 	}
 }
 
+func TestLogrusToZapLevel(t *testing.T) {
+	tests := []struct {
+		name           string
+		zapConfig      *zap.Config
+		levelsEnabled  []zapcore.Level
+		levelsDisabled []zapcore.Level
+	}{
+		{
+			name: "debug",
+			zapConfig: &zap.Config{
+				Level: zap.NewAtomicLevelAt(zap.DebugLevel),
+			},
+			levelsEnabled: zapLevels,
+		},
+		{
+			name: "info",
+			zapConfig: &zap.Config{
+				Level: zap.NewAtomicLevelAt(zap.InfoLevel),
+			},
+			levelsEnabled: []zapcore.Level{
+				zapcore.InfoLevel,
+				zapcore.WarnLevel,
+				zapcore.ErrorLevel,
+				zapcore.DPanicLevel,
+				zapcore.PanicLevel,
+				zapcore.FatalLevel,
+			},
+			levelsDisabled: []zapcore.Level{
+				zapcore.DebugLevel,
+			},
+		},
+		{
+			name: "warn",
+			zapConfig: &zap.Config{
+				Level: zap.NewAtomicLevelAt(zap.WarnLevel),
+			},
+			levelsEnabled: []zapcore.Level{
+				zapcore.WarnLevel,
+				zapcore.ErrorLevel,
+				zapcore.DPanicLevel,
+				zapcore.PanicLevel,
+				zapcore.FatalLevel,
+			},
+			levelsDisabled: []zapcore.Level{
+				zapcore.InfoLevel,
+				zapcore.DebugLevel,
+			},
+		},
+		{
+			name: "error",
+			zapConfig: &zap.Config{
+				Level: zap.NewAtomicLevelAt(zap.ErrorLevel),
+			},
+			levelsEnabled: []zapcore.Level{
+				zapcore.ErrorLevel,
+				zapcore.DPanicLevel,
+				zapcore.PanicLevel,
+				zapcore.FatalLevel,
+			},
+			levelsDisabled: []zapcore.Level{
+				zapcore.InfoLevel,
+				zapcore.WarnLevel,
+				zapcore.DebugLevel,
+			},
+		},
+		{
+			name: "dpanic",
+			zapConfig: &zap.Config{
+				Level: zap.NewAtomicLevelAt(zap.DPanicLevel),
+			},
+			levelsEnabled: []zapcore.Level{
+				zapcore.DPanicLevel,
+				zapcore.PanicLevel,
+				zapcore.FatalLevel,
+			},
+			levelsDisabled: []zapcore.Level{
+				zapcore.DebugLevel,
+				zapcore.InfoLevel,
+				zapcore.WarnLevel,
+				zapcore.ErrorLevel,
+			},
+		},
+		{
+			name: "panic",
+			zapConfig: &zap.Config{
+				Level: zap.NewAtomicLevelAt(zap.PanicLevel),
+			},
+			levelsEnabled: []zapcore.Level{
+				zapcore.PanicLevel,
+				zapcore.FatalLevel,
+			},
+			levelsDisabled: []zapcore.Level{
+				zapcore.DebugLevel,
+				zapcore.InfoLevel,
+				zapcore.WarnLevel,
+				zapcore.ErrorLevel,
+				zapcore.DPanicLevel,
+			},
+		},
+		{
+			name: "fatal",
+			zapConfig: &zap.Config{
+				Level: zap.NewAtomicLevelAt(zap.FatalLevel),
+			},
+			levelsEnabled: []zapcore.Level{
+				zapcore.FatalLevel,
+			},
+			levelsDisabled: []zapcore.Level{
+				zapcore.DebugLevel,
+				zapcore.InfoLevel,
+				zapcore.WarnLevel,
+				zapcore.ErrorLevel,
+				zapcore.PanicLevel,
+				zapcore.DPanicLevel,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			test.zapConfig.Encoding = "json"
+			logger, err := test.zapConfig.Build()
+			require.NoError(t, err)
+
+			logToZap := newLogrusToZap(func() *zap.Logger { return logger })
```

Comment:
Why aren't we using the real `loggerProvider` with `newDefaultLoggerCfg` call?  I think otherwise this test confirms that `logToZap.defaultLogger` is the same logger instance, which we don't want.

Reply:
I think this would also help the patch coverage

Reply:
Ah good catch, fixed.